k8s/jenkins/legion-jenkins-plugin/src/main/resources/legion.groovy

```diff
@@ -48,28 +66,49 @@ def modelFileName() {
 }
 
 def defaultModelId(notebookName) {
+    /**
+     * Get model id from file name
+     */
     return (notebookName.indexOf('.') >= 0) ?
             notebookName.substring(0, notebookName.lastIndexOf('.')) : notebookName
 }
 
 def getDefaultImageName(){
+    /**
+     * Get default base image name
+     */
     return System.getenv("LEGION_BASE_IMAGE_REPOSITORY") + ":" + System.getenv("LEGION_BASE_IMAGE_TAG")
 }
 
 def pod(Map podParams=null, Closure body) {
+    /**
+     * Run closure in jenkins slave on K8S pod
+     */
+
+    // Default arguments
     if (podParams == null)
         podParams = [:]
     
-    if ( params.Enclave == "" ) {
+    // Parse parameters
+    image = podParams.get('image', getDefaultImageName())
+    cpu = podParams.get('cpu', '330m')
+    ram = podParams.get('ram', '4Gi')
+    enclave = podParams.get('enclave', '')
+
+    // Override enclave if parameter has been passed
+    if (params.Enclave)
+        enclave = params.Enclave
+
+    // Check enclave parameter
+    if (!enclave){
         echo '[FAILURE] Parameter Enclave not defined!'
         currentBuild.result = 'FAILURE'
         return
     }
+    else
+        println "Using enclave ${enclave}"
 
-    image = podParams.get('image', getDefaultImageName())
-    cpu = podParams.get('cpu', '330m')
-    ram = podParams.get('ram', '4Gi')
-
+    // Create list of env variables that should be passed to slave
     envToPass = [
```

Comment:
@kirillmakhonin the list of env vars should be defined in configuration file but not hardcoded in groovy lib script